pkg/util/ipsec/ipsec_telemetry.go

```diff
@@ -0,0 +1,51 @@
+package ipsec
+
+import (
+	"sync"
+
+	operv1 "github.com/openshift/api/operator/v1"
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+	"k8s.io/klog/v2"
+)
+
+var (
+	registerMetrics sync.Once
+	ipsecStateGauge *metrics.GaugeVec
+
+	recordedMode, recordedFlavor string = "", ""
+)
+
+// RegisterIpsecMetrics in the global registry
+func RegisterIpsecMetrics() {
+	registerMetrics.Do(func() {
+		ipsecStateGauge = metrics.NewGaugeVec(&metrics.GaugeOpts{
+			Name: "cluster_network_ipsec_state",
+			Help: "A metric with a constant '1' value labeled by the latest ipsecMode of the cluster",
+		}, []string{"mode", "api_flavor"})
+		legacyregistry.MustRegister(ipsecStateGauge)
+	})
+}
+
+func UpdateIpsecTelemetry(ipsecConfig *operv1.IPsecConfig) {
+	var flavor, mode string
+	klog.V(5).Infof("IPsec Telemetry: %v", ipsecConfig)
+	RegisterIpsecMetrics()
+	if ipsecConfig == nil {
+		mode = string(operv1.IPsecModeDisabled)
+		flavor = "4.14"
+	} else if ipsecConfig.Mode == "" {
+		mode = string(operv1.IPsecModeFull)
+		flavor = "4.14"
+	} else {
+		mode = string(ipsecConfig.Mode)
+		flavor = "4.15"
```

Comment:
what's the significance of exposing flavor in the metric ? I'm not sure if it's good idea to hardcode these values with 4.15 or 4.14. Did you consider to use https://github.com/openshift/cluster-network-operator/blob/master/vendor/github.com/openshift/api/operator/v1/types.go#L123 ?

Reply:
I am not sure I understand this - we have 2 versions of the API, and they changed between 4.14 and 4.15. I believe it is good to know if someone is still using the older version. I can call it "old" and "new", or any other agreed constant names, but if we collect telemetry, it is an important piece of information. I don't understand how the `Version` field can help me - since I need constants and not something that can change according to the version in which the code is executed.

Reply:
This is an option - I am not sure if booleans are supported by Prometheus labels, but it can be "true" or "false" if not. I will do it.